advisories/hackage/cryptonite/HSEC-2025-0002.md

```diff
@@ -0,0 +1,41 @@
+```toml
+[advisory]
+id = "HSEC-2025-0002"
+cwe = []
+keywords = ["crypto", "historical"]
+related = ["GHSA-w5vr-6qhr-36cc"]
+
+[[affected]]
+package = "cryptonite"
+[[affected.versions]]
+introduced = "0.1.0"
+
+[[references]]
+type = "ARTICLE"
+url = "https://portswigger.net/daily-swig/dozens-of-cryptography-libraries-vulnerable-to-private-key-theft"
+[[references]]
+type = "ADVISORY"
+url = "https://github.com/advisories/GHSA-w5vr-6qhr-36cc"
+[[references]]
+type = "EVIDENCE"
+url = "https://hackage.haskell.org/package/cryptonite-0.30/docs/src/Crypto.PubKey.Ed25519.html#ccryptonite_ed25519_sign"
+
+```
+
+# Double Public Key Signing Function Oracle Attack on Ed25519
+
+The standard specification of Ed25519 message signing involves providing the
+algorithm with a message and private key.
+
+The function will use the private key to compute the public key and sign the message.
+Some libraries provide a variant of the message signing function that also takes
+the pre-computed public key as an input parameter.
+
+Some libraries were allowing arbitrary public keys as inputs without checking
```

Comment:
Perhaps this could be rephrased as "Libraries that allow arbitrary public keys as inputs without checking ... are vulnerable to the following attack"

Then below, we could use this format: "By doing [action] a [actor] may [impact] resulting in [consequence].", e.g.:

```
By using several public keys and messages, a malicious user with access to the signing mechanism
may build up insights into the private key parameters resulting in access to the private key.
```

Reply:
Fixed, thanks.